test/test_common/utility.cc

```diff
@@ -156,6 +156,37 @@ envoy::api::v2::Bootstrap TestUtility::parseBootstrapFromJson(const std::string&
   return bootstrap;
 }
 
+std::vector<std::string> TestUtility::split(const std::string& source, char split) {
+  return TestUtility::split(source, std::string{split});
+}
+
+std::vector<std::string> TestUtility::split(const std::string& source, const std::string& split,
+                                            bool keep_empty_string) {
+  std::vector<std::string> ret;
+  size_t last_index = 0;
+  size_t next_index;
+
+  if (split.empty()) {
+    ret.emplace_back(source);
+    return ret;
+  }
+
+  do {
+    next_index = source.find(split, last_index);
+    if (next_index == std::string::npos) {
+      next_index = source.size();
+    }
+
+    if (next_index != last_index || keep_empty_string) {
+      ret.emplace_back(source.substr(last_index, next_index - last_index));
+    }
+
+    last_index = next_index + split.size();
+  } while (next_index != source.size());
```

Comment:
next_index < source.size()

it's hard for me to be sure that you won't shoot past source.size() given that you assign next_index to source.size() on line 177, and then ast last_index=next_index+split.size() above.

this makes me wonder how well this function is tested.  Maybe you could redefine it in terms of absl::StrSplit and then convert the result to an array of std::string?

Reply:
This is a test function. Can't we just call the string_view version and then just convert the result into an array of strings?

Reply:
This is the old split function. I just moved it into common_test_lib for convenience when we are writing tests. I can try to call the string_view version and then transform the result.